lib/components/molecules/Accordion/Accordion.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+/**
+ *
+ * Accordion
+ *
+ */
+import React, { useState } from 'react';
+import type { Node } from 'react';
+import styled from 'styled-components';
+import classnames from 'classnames';
+import styles from './Accordion.style';
+import type { AccordionHeaderProps, AccordionPanelProps, AccordionProps } from './types';
+
+const AccordionHeader = ({
+  children,
+  className,
+  ariaLabel,
+  accordionPanelId,
+}: AccordionHeaderProps): Node => {
+  const [isExpanded, toggleExpanded] = useState(false);
```

Comment:
What you did here is functionally right but this should have happened on the AccordionPanel Component and from here you will only fire the clickHandler passed from AccordionPanel Component. The state will be passed down from AccordionPanel and this component when clicked just toggle it and send into the action to be updated in the parent component. 

Please understand the Detail is collapsing not the summary. So state should be maintained by AccordionPanel